src/js/components/DataTable/Header.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+
+import { TableCell } from '../Table';
+import { Text } from '../Text';
+import { Box } from '../Box';
+
+import Resizer from './Resizer';
+import Searcher from './Searcher';
+import Sorter from './Sorter';
+import ExpanderCell from './ExpanderCell';
+import { StyledDataTableHeader, StyledDataTableRow } from './StyledDataTable';
+
+const Header = ({
+  columns, filtering, filters, groups, groupState,
+  onFilter, onFiltering, onResize, onSort, onToggle,
+  sort, theme, widths, ...rest
+}) => (
+  <StyledDataTableHeader {...rest}>
+    <StyledDataTableRow>
+
+      {groups ? (
+        <ExpanderCell
+          context='header'
+          expanded={Object.keys(groupState)
+            .filter(k => !groupState[k].expanded).length === 0}
+          theme={theme}
+          onToggle={onToggle}
+        />
+      ) : null}
+
+      {columns.map(({ property, header, align, search }) => (
+        <TableCell
+          key={property}
+          scope='col'
+          plain={true}
+          verticalAlign='bottom'
+          style={widths && widths[property] ?
```

Comment:
https://github.com/grommet/grommet/blob/NEXT/src/js/components/Select/SelectContainer.js#L21

Reply:
I don't think it makes sense to use that strategy here, as this is where the caller is resizing the column widths interactively.

Reply:
agreed